landlab/data_record/data_record.py

```diff
@@ -347,7 +383,11 @@ def _check_element_id_values(self, grid_element, element_id):
             )
 
     def add_record(self, time=None, item_id=None, new_item_loc=None, new_record=None):
-        """Add a time-related record to an existing DataRecord.
+        """Add a new record to an existing DataRecord.
+
+        Unlike add_item, this method can support adding records that include
+        new variables to the DataRecord. It can also support adding records
+        that do not include time.
```

Comment:
Similar to in the __init__ I've added text here trying to distinguish between add_record and add_item. Please comment/improve/etc.